esiclient/utils.py

```diff
@@ -25,33 +25,39 @@ def get_network_display_name(network):
         return network.name
 
 
-def get_network_info_from_port(port, client):
+def get_network_info_from_port(port, client, networks_dict={}):
     """Return Neutron network name and ips from port
 
     :param port: a Neutron port
+    :param networks_dict: networks dict {id:network}
     :param client: neutron client
     """
-    network = client.get_network(port.network_id)
+    if networks_dict and port.network_id in networks_dict:
```

Comment:
Suuuuper minor - you no longer need the first part of this conditional, because the function will always define a networks_dict